src/components/People/People.tsx

```diff
@@ -0,0 +1,164 @@
+/* eslint-disable no-nested-ternary */
+import { useEffect, useState } from 'react';
+import { useHistory, useLocation } from 'react-router-dom';
+import { Person } from '../../types';
+import { Loader } from '../Loader';
+import { getPeople } from '../../api';
+
+export const People = () => {
+  const { pathname } = useLocation<{ pathname: string }>();
+  const [people, setPeople] = useState<Person[]>([]);
+  const [errorMessage, setErrorMessage] = useState('');
+  const [slugUser, setSlugUser] = useState(pathname.slice(8));
+  const [isLoading, setIsLoading] = useState(false);
+
+  useEffect(() => {
+    const fetchPeople = async () => {
+      setIsLoading(true);
+      try {
+        const data = await getPeople();
+
+        setPeople(data);
+      } catch (error) {
+        setErrorMessage('Unable to load people');
+      }
+
+      setIsLoading(false);
+    };
+
+    fetchPeople();
+  }, []);
+
+  const history = useHistory();
+
+  useEffect(() => {
+    if (!slugUser) {
+      history.push('/people');
+    }
+
+    history.push(`/people/${slugUser}`);
+  }, [slugUser]);
+
+  const findParent = (person: Person) => {
+    let parent = people.find(one => one.name === person.motherName);
+
+    if (!parent) {
+      parent = people.find(one => one.name === person.fatherName);
+    }
+
+    if (!parent) {
+      return person;
+    }
+
+    return parent;
+  };
+
+  return (
+    <div className="block">
+      <div className="box table-container">
+        {isLoading && (<Loader />)}
+
+        {errorMessage && (
+          <>
+            <p
+              data-cy="peopleLoadingError"
+              className="has-text-danger"
+            >
+              Something went wrong
+            </p>
+            <p data-cy="noPeopleMessage">
+              There are no people on the server
+            </p>
+          </>
+        )}
+
+        <table
```

Comment:
From the task requirements you should create separate component, e.g. `/components/PeopleTable.tsx` and pass props data there.

"Pass people to PeopleTable component and render as a table (see the given markup);"

it can be something like this:

```
<PeopleTable
        people={people}
        errorMessage={errorMessage}
        isLoading={isLoading}
      />
```